### PR TITLE
Use more intra-doc-links in `core::fmt`

### PR DESCRIPTION
**Pull request by LeSeulArtichaut**
This is a follow-up to #75819, which encountered some broken links due to #75176, so this PR contains the links that are blocked on #75176.

r? @jyn514

### CODE REVIEW

**Comment by jyn514** [comment]
This can be merged once you rebase :) assuming that `--stage 0` doesn't error which it might. If so it will have to wait another 5 weeks :/

**Comment by camelid** [comment]
triage: Is this still blocked?

File: library/core/src/fmt/mod.rs [hunk 1]
```diff
@@ -102,7 +102,7 @@ pub struct Error;
 /// library. The [`write!`] macro accepts an instance of [`io::Write`], and the
 /// [`io::Write`] trait is favored over implementing this trait.
 ///
-/// [`write!`]: ../../std/macro.write.html
+/// [`write!`]: crate::write!
```
Comment (camelid) [inline library/core/src/fmt/mod.rs:105]: Isn't this auto-imported?

```suggestion
```

File: library/core/src/fmt/mod.rs [hunk 2]
```diff
@@ -1058,7 +1058,7 @@ pub trait UpperExp {
 /// assert_eq!(output, "Hello world!");
 /// ```
 ///
-/// [`write!`]: ../../std/macro.write.html
+/// [`write!`]: crate::write!
```
Comment (camelid) [inline library/core/src/fmt/mod.rs:1061]: Same here.

```suggestion
```

File: library/core/src/fmt/mod.rs [hunk 3]
```diff
@@ -1886,7 +1886,7 @@ impl<'a> Formatter<'a> {
     /// assert_eq!(format!("{:?}", Foo(vec![10, 11])), "{10, 11}");
     /// ```
     ///
-    /// [`format_args!`]: ../../std/macro.format_args.html
+    /// [`format_args!`]: crate::format_args
     ///
```
Comment (camelid) [inline library/core/src/fmt/mod.rs:1890]: This too:

```suggestion
```

**Review by camelid — CHANGES_REQUESTED** [review]
It seems like all of these are auto-imported, which might resolve the blockage?

**Comment by jyn514** [comment]
This should be ready to go after a rebase :)

**Comment by LeSeulArtichaut** [comment]
Rebased

**Review by jyn514 — APPROVED** [review]
It works :tada:

**Comment by camelid** [comment]
Aren't all these links unnecessary though, since they're in the prelude?

**Comment by jyn514** [comment]
Sure, we can fix them in a follow-up. I mostly want to finally get this merged 😆

**Pull request merged**